mbed_devices/_internal/windows/component_descriptor.py

```diff
@@ -0,0 +1,123 @@
+"""Defines a generic Win32 component."""
+import logging
+from abc import ABC, abstractmethod
+from typing import List, Any, Generator, Optional
+
+import win32com.client
+
+UNKNOWN_VALUE = "Unknown"
+NAMED_TUPLE_FIELDS_ATTRIBUTE = "_fields"
+
+logger = logging.getLogger(__name__)
+
+
+class ComponentDescriptor(ABC):
+    """Win32 component descriptor."""
+
+    @property
+    @abstractmethod
+    def class_name(self) -> str:
+        """Returns the name of the Win32 Class."""
+        return UNKNOWN_VALUE
+
+    @property
+    def parameters(self) -> List[str]:
+        """Returns all the different parameters of the class."""
+        return self._class_parameters()
+
+    @classmethod
+    def _class_parameters(cls) -> List[str]:
```

Comment:
This shouldn't be a `@classmethod` I'd expect it to return an instance of `cls` where this returns a list of class attributes.